src/main/java/dal/fatih/todoly/HibernateTaskRepository.java

```diff
@@ -0,0 +1,79 @@
+package dal.fatih.todoly;
+
+
+import javax.persistence.*;
+import java.sql.Date;
+import java.util.*;
+
+public class HibernateTaskRepository implements TaskRepository {
+
+    private final EntityManagerFactory entityManagerFactory
+            = Persistence.createEntityManagerFactory("Todoly");
+    private final EntityManager entityManager
+            = entityManagerFactory.createEntityManager();
+
+    @Override
+    public boolean create(Task task) {
+        entityManager.getTransaction().begin();
+        entityManager.persist(task);
+        entityManager.getTransaction().commit();
+
+        return true;
+    }
+
+    @Override
+    public List<Task> list() {
+        final TypedQuery<Task> listQuery
+                = entityManager.createQuery("select a from Task a ", Task.class);
+        return listQuery.getResultList();
+    }
+
+    @Override
+    public Task get(String taskId) {
+        final TypedQuery<Task> getQuery
+                = entityManager.createQuery("select a from Task a " +
+                "where a.taskId ='" + taskId + "'", Task.class);
+        return getQuery.getSingleResult();

```

Comment:
Use hibernate criteria API. Hibernate is also for having as little SQL as possible. https://www.tutorialspoint.com/jpa/jpa_criteria_api.htm#:~:text=The%20Criteria%20API%20is%20a,modify%20by%20changing%20the%20syntax.